forage-android/src/main/java/com/joinforage/forage/android/vault/BasisTheoryPinSubmitter.kt

```diff
@@ -127,5 +134,30 @@ internal class BasisTheoryPinSubmitter(
                 .apiKey(API_KEY)
                 .build()
         }
+
+        /**
+         * com.basistheory.ApiException isn't currently
+         * publicly-exposed by the basis-theory-android package
+         * so we parse the raw exception message to retrieve the body of the BasisTheory
+         * errors
+         *
+         * @return the BasisTheory JSON response body string if the response is a BasisTheory error,
+         * or null if it is not
+         */
+        private fun getBasisTheoryExceptionBody(vaultResponse: BasisTheoryResponse): String? {
+            val rawBasisTheoryError = vaultResponse.exceptionOrNull()?.message ?: return null
+            val bodyRegex = "HTTP response body: (.+?)\\n".toRegex()
+            return bodyRegex.find(rawBasisTheoryError)?.groupValues?.get(1)
+        }
```

Comment:
I reached out to the Basis Theory team about this limitation with ApiException.

It is still safe to release these changes as-is